### PR TITLE
[move] View APIs instead of Move resolvers

### PR DESCRIPTION
**Pull request by georgemitenkov**
## Description

Previously, `MoveResolver`s have been used to access storage bytes by the VM, Move APIs, indexer, etc. As a result, this created an unnecessary coupling between the VM and the rest of the system, making code maintainability harder, error-prone and hard to evolve (e.g., implementing a new loader).

This PR fixes that by:
- Ensuring `MoveResolver`s are not used outside of the VM. This way VM team can evolve the storage accessing API separately, not affecting indexing and testing logic.
- Using `AptosValueAnnotator` over `MoveValueAnnotator` where possible
- Renaming `GetModule` trait to cache compiled module over bytes, and moving the caching logic from `MoveValueAnnotator` to the outer layer.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [x] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [x] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

Existing tests.

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: third_party/move/tools/move-bytecode-utils/src/compiled_module_viewer.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+use move_binary_format::CompiledModule;
+use move_core_types::language_storage::ModuleId;
+use std::borrow::Borrow;
+
+pub trait CompiledModuleView {
+    type Item: Borrow<CompiledModule>;
+
+    fn view_compiled_module(&self, id: &ModuleId) -> anyhow::Result<Self::Item>;
```
Comment (vgao1996) [inline third_party/move/tools/move-bytecode-utils/src/compiled_module_viewer.rs:12]: Note: consider taking addr & module name as separate arguments to allow more clients to use this API

File: aptos-move/aptos-resource-viewer/src/lib.rs [hunk 2]
```diff
@@ -2,54 +2,73 @@
 // Parts of the project are originally copyright © Meta Platforms, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use anyhow::{bail, Result};
-use aptos_types::{access_path::AccessPath, contract_event::ContractEvent};
-use move_core_types::{language_storage::StructTag, resolver::ModuleResolver};
+pub mod module_view;
```
Comment (vgao1996) [inline aptos-move/aptos-resource-viewer/src/lib.rs:5]: Would love to see this helper library being documented.

File: api/src/accounts.rs [hunk 3]
```diff
@@ -519,17 +516,20 @@ impl Account {
     /// Find a resource associated with an account
     fn find_resource(
         &self,
-        resource_type: &StructTag,
+        tag: &StructTag,
```
Comment (vgao1996) [inline api/src/accounts.rs:519]: Nit: `resource_type` or `value_type` would be more meaningful than `tag`

**Review by vgao1996 — APPROVED** [review]
LGTM, but please make sure the API folks have a chance to look at this

File: api/src/accounts.rs (hunk 3 above)
Comment (georgemitenkov) [reply api/src/accounts.rs:519]: Done

File: third_party/move/tools/move-bytecode-utils/src/compiled_module_viewer.rs (hunk 1 above)
Comment (georgemitenkov) [reply third_party/move/tools/move-bytecode-utils/src/compiled_module_viewer.rs]: I think we should just use addr & module name for this API, should be sufficient. I added a note and will make a pass after this PR lands

**Review by runtian-zhou — APPROVED** [review]

**Pull request merged**